### DISCUSSION TITLE
Precedence Matching

### DISCUSSION THREAD

**Discussion by realsama** [pest-parser/pest#723] [Q&A] [answered]
Hello, I am trying to parse a script with constructs like arithmetic operations, functions and lambdas. I know that I can use [PrattParser](https://docs.rs/pest/latest/pest/pratt_parser/struct.PrattParser.html) to parse the arithmetic logic and do precedence climbing, and I also know that I can just use ```MyParser::pars``` to parse other constructs. 

My question is how do I combine these two objectives into a single parsing program? Thanks

**Comment by tomtau** [comment, marked as answer]
I tried to complete these missing book sections:

https://pest-parser.github.io/book/precedence.html
https://pest-parser.github.io/book/examples/calculator.html

The Calculator example is based on the excellent tutorial originally written by @wildarch. For questions or suggestions, please feel free to open issues or submit a PR on https://github.com/pest-parser/book !

**Reply by realsama** [reply]
> https://pest-parser.github.io/book/examples/calculator.html

Thank you for updating the book.  Looks like you missed content of the pest grammar file for the pest calculator module on github [here](https://github.com/pest-parser/book/blob/master/examples/pest-calculator/src/calculator.pest).

**Reply by tomtau** [reply]
Good catch, thanks! The grammar file should be there now.

**Reply by realsama** [reply]
Seen. Thank you!
